test/OrchardCore.Tests/Apis/Context/AuthenticationContext.cs

```diff
@@ -14,11 +16,15 @@ internal class PermissionContextAuthorizationHandler : AuthorizationHandler<Perm
         private readonly PermissionsContext _permissionsContext;
 
         // Used for http based graphql tests, marries a permission context to a request
-        public PermissionContextAuthorizationHandler(IHttpContextAccessor httpContextAccessor, IDictionary<string, PermissionsContext> permissionsContexts)
+        public PermissionContextAuthorizationHandler(
+            IServiceProvider serviceProvider,
+            IHttpContextAccessor httpContextAccessor,
+            IDictionary<string, PermissionsContext> permissionsContexts)
         {
             _permissionsContext = new PermissionsContext();
 
-            var requestContext = httpContextAccessor.HttpContext.Request;
+            var httpContext = serviceProvider.GetRequiredService<IHttpContextAccessor>().HttpContext;
```

Comment:
This is also injected.

Reply:
What do you mean by that? What should I do here?

Reply:
That `IContextAccessor `is a mocked one. I only use it if this injected `httpContext `is null. I tried to use only the mocked `IContextAccessor`, but the GraphQLClient doesn't work with that.

Reply:
When `PermissionContextAuthorizationHandler` is resolved (instantiated) the `IHttpContextAccessor` it gets injected into its ctor is the same that's in the injected `IServiceProvider` that you're resolving here (since that `IServiceProvider` is what `PermissionContextAuthorizationHandler` is also resolved from). So, there shouldn't be any difference, because you're accessing the same service from the same Di container.

Reply:
I understand what you are saying, but the `IHttpContextAccessor` in the ctor is a mock object.

![image](https://user-images.githubusercontent.com/79921246/162455207-9dad7549-ac57-4f56-bb05-5609e5f29c78.png)

Reply:
I see. `PermissionContextAuthorizationHandler` should use the same service in all cases. If you need to override what gets into the container, you can register the mock instance in `SiteStartup`.